src/plugins/vis_type_timeseries/common/timefield_utils.ts

```diff
@@ -17,12 +17,16 @@
  * under the License.
  */
 
-import _ from 'lodash';
+import { TimeFieldObject } from './types';
 
-export const createSelectHandler = (handleChange) => {
-  return (name) => (selectedOptions) => {
-    return handleChange?.({
-      [name]: _.get(selectedOptions, '[0].value', null),
-    });
-  };
+const isTimefieldObject = (
+  timeField: TimeFieldObject
+): timeField is { name: string; label: string } => {
+  return Boolean(timeField && typeof timeField !== 'string' && timeField.name);
 };
+
+export const extractTimefieldName = (timeField: TimeFieldObject) =>
```

Comment:
I think these functions name should not include `timeField` since it is used for all field types?